### PR TITLE
Simulator

### PR DESCRIPTION
**Pull request by ANeaves**
Created Frame Simulator Plugin for Mercury. Currently has either the option to play back pcap files, or generate frames of data with incrementing pixel values. Further test patterns can be added in future work if desired

### CODE REVIEW

File: data/cmake/FindOdinData.cmake [hunk 1]
```diff
@@ -46,9 +46,11 @@ ENDIF(PkgConfig_FOUND)
 if (NOT APPLE)
     set(LIB_RECEIVER FrameReceiver)
     set(LIB_PROCESSOR FrameProcessor)
+	set(LIB_SIMULATOR FrameSimulator)
 else(NOT APPLE)
```
Comment (dw96) [inline data/cmake/FindOdinData.cmake:50]: Are lines 49 and 53 indented to the same level as the rest of their respective paragraphs? In the browser it doesn't appear to be that way. Nitpicking: Line 53 "SET" in capitals whereas the above two lines the same keyword is in lower case.

File: data/frameSimulator/src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+set(CMAKE_INCLUDE_CURRENT_DIR on)
+ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK)
+
+include_directories(${FRAMESIMULATOR_DIR}/include ${ODINDATA_INCLUDE_DIRS}
+	${Boost_INCLUDE_DIRS} ${LOG4CXX_INCLUDE_DIRS}/.. ${ZEROMQ_INCLUDE_DIRS} ${PCAP_INCLUDE_DIR})
```
Comment (dw96) [inline data/frameSimulator/src/CMakeLists.txt]: Nitpicking but: should "Boost_INCLUDE_DIRS" be all capitals i.e. "BOOST_INCLUDE_DIRS"?
I suspect that CMake is case insensitive though.

File: data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+#include <iostream>
+#include <algorithm>
+#include <boost/lexical_cast.hpp>
+
+#include "version.h"
+
+namespace FrameSimulator {
+
+
+    /** Construct an MercuryFrameSimulatorPlugin
```
Comment (dw96) [inline data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp]: Feel free to ignore this one as it has nothing to do with the code but "an Mercury.." should be "a Mercury..".

Is there a grammar Nazi emoticon? :-p

**Review by dw96 — APPROVED** [review]
Mostly just some formatting comments, as far as I can tell the source code itself is fine.

File: data/cmake/FindOdinData.cmake (hunk 1 above)
Comment (ANeaves) [reply data/cmake/FindOdinData.cmake:50]: For some reason, VS Code doesn't do the normal "put 4 spaces in when you hit the tab key" thing when editing cmake files, hence the odd looking indentation. Fixed it

File: data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp (hunk 3 above)
Comment (timcnicholls) [reply data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp]: I don't think pointing out a typo qualifies you as a grammar Nazi 🤷‍♂️ 👍  

Guess this is a hangover from EXCALIBUR, HEXITEC, ....

File: data/frameSimulator/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Add the src subdirectory for frameSimulator
+add_subdirectory(src)
```
Comment (timcnicholls) [inline data/frameSimulator/CMakeLists.txt]: Missing newline at end of file

File: data/frameSimulator/include/MercuryFrameSimulatorPlugin.h [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        int total_bytes;
+
+        int current_frame_num;
+
+    };
+
+    REGISTER(FrameSimulatorPlugin, MercuryFrameSimulatorPlugin, "MercuryFrameSimulatorPlugin");
+}
+
+#endif //FRAMESIMULATOR_MERCURYFRAMESIMULATORPLUGIN_H
```
Comment (timcnicholls) [inline data/frameSimulator/include/MercuryFrameSimulatorPlugin.h]: Missing newline at end of file

**Review by timcnicholls — COMMENTED** [review]
Looks good - couple of minor formatting commments but otherwise no probs. 👍

**Pull request merged**